From 1469973c4cc5abe29b5c6abb31d5433eaf3e92d7 Mon Sep 17 00:00:00 2001
From: Darek Stojaczyk <dariusz.stojaczyk@intel.com>
Date: Fri, 22 May 2020 17:15:28 +0200
Subject: [PATCH 1/3] build/meson: disable libraries we don't need

Only the following are left, all of which will be built
unconditionally:

// core
LIBS="eal ring mempool pci "
// eal dependencies
LIBS+="kvargs telemetry "

// governor
LIBS+="power timer ethdev net"

// crypto & compress deps
LIBS+="reorder cryptodev "
LIBS+="compressdev "
LIBS+="security "

// vhost and its dependencies
LIBS+="cryptodev mbuf cmdline meter hash vhost "

// raid5 depencency
LIBS+="hash rcu"

Also disable building any DPDK apps and make compress_isal PMD
work without the system-wide libisal. We'll provide our isa-l
submodule with -I and -L. Simply remove the security checks
from meson file and attempt building the PMD whenever requested.

spdk-20.08 note: added building 'security' as this became required for
aesni_mb and qat. Squashed patches disabling 'graph'/'node' and reenabling
'telemetry'. Disabled "regexdev" lib.

spdk-20.11 note: added building 'rte_power' required for SPDK governor.
Added rte_rcu as rte_hash dependcy starting with DPDK 20.11.

spdk-21.02 note: moved ethdev and net as requirements for SPDK governor.

Change-Id: I036b84178563c9b165db2597bf244bf59e238662
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/2578 (spdk-19.11.2)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/3352 (spdk-20.05)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/3769 (spdk-20.08)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/5337 (spdk-20.11)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/6427 (spdk-21.02)
Signed-off-by: Darek Stojaczyk <dariusz.stojaczyk@intel.com>
Signed-off-by: Tomasz Zawadzki <tomasz.zawadzki@intel.com>
Signed-off-by: Jim Harris <james.r.harris@intel.com>
Signed-off-by: Krzysztof Karas <krzysztof.karas@intel.com>
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/8135
Reviewed-by: Aleksey Marchuk <alexeymar@mellanox.com>
Reviewed-by: Ben Walker <benjamin.walker@intel.com>
---
 drivers/compress/isal/meson.build |  4 ++--
 lib/acl/meson.build               |  2 ++
 lib/bbdev/meson.build             |  2 ++
 lib/bitratestats/meson.build      |  2 ++
 lib/bpf/meson.build               |  2 ++
 lib/cfgfile/meson.build           |  2 ++
 lib/distributor/meson.build       |  2 ++
 lib/efd/meson.build               |  2 ++
 lib/eventdev/meson.build          |  2 ++
 lib/fib/meson.build               |  2 ++
 lib/flow_classify/meson.build     |  2 ++
 lib/graph/meson.build             |  2 ++
 lib/gro/meson.build               |  2 ++
 lib/gso/meson.build               |  2 ++
 lib/ip_frag/meson.build           |  2 ++
 lib/ipsec/meson.build             |  2 ++
 lib/jobstats/meson.build          |  2 ++
 lib/kni/meson.build               |  2 ++
 lib/latencystats/meson.build      |  2 ++
 lib/lpm/meson.build               |  2 ++
 lib/member/meson.build            |  2 ++
 lib/metrics/meson.build           |  2 ++
 lib/node/meson.build              |  2 ++
 lib/pdump/meson.build             |  2 ++
 lib/pipeline/meson.build          |  2 ++
 lib/port/meson.build              |  2 ++
 lib/rawdev/meson.build            |  2 ++
 lib/regexdev/meson.build          |  2 ++
 lib/rib/meson.build               |  2 ++
 lib/sched/meson.build             |  2 ++
 lib/stack/meson.build             | 11 ++++++-----
 lib/table/meson.build             |  2 ++
 meson.build                       |  2 +-
 33 files changed, 69 insertions(+), 8 deletions(-)

diff --git a/drivers/compress/isal/meson.build b/drivers/compress/isal/meson.build
index 4b3eaa2274..22eb77c685 100644
--- a/drivers/compress/isal/meson.build
+++ b/drivers/compress/isal/meson.build
@@ -3,8 +3,8 @@
 
 dep = dependency('libisal', required: false, method: 'pkg-config')
 if not dep.found()
-    build = false
-    reason = 'missing dependency, "libisal"'
+	build = true
+	reason = 'missing dependency, "libisal"'
 endif
 
 deps += 'bus_vdev'
diff --git a/lib/acl/meson.build b/lib/acl/meson.build
index fbb1318230..4f44537487 100644
--- a/lib/acl/meson.build
+++ b/lib/acl/meson.build
@@ -81,3 +81,5 @@ elif dpdk_conf.has('RTE_ARCH_ARM')
 elif dpdk_conf.has('RTE_ARCH_PPC_64')
     sources += files('acl_run_altivec.c')
 endif
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/bbdev/meson.build b/lib/bbdev/meson.build
index 2969cab26b..7a1832dbc6 100644
--- a/lib/bbdev/meson.build
+++ b/lib/bbdev/meson.build
@@ -6,3 +6,5 @@ headers = files('rte_bbdev.h',
         'rte_bbdev_pmd.h',
         'rte_bbdev_op.h')
 deps += ['mbuf']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/bitratestats/meson.build b/lib/bitratestats/meson.build
index ede7e0a579..081552048c 100644
--- a/lib/bitratestats/meson.build
+++ b/lib/bitratestats/meson.build
@@ -4,3 +4,5 @@
 sources = files('rte_bitrate.c')
 headers = files('rte_bitrate.h')
 deps += ['ethdev', 'metrics']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/bpf/meson.build b/lib/bpf/meson.build
index 63cbd60185..35371ba99a 100644
--- a/lib/bpf/meson.build
+++ b/lib/bpf/meson.build
@@ -25,3 +25,5 @@ if dep.found()
     sources += files('bpf_load_elf.c')
     ext_deps += dep
 endif
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/cfgfile/meson.build b/lib/cfgfile/meson.build
index 88eb819856..1f48577180 100644
--- a/lib/cfgfile/meson.build
+++ b/lib/cfgfile/meson.build
@@ -3,3 +3,5 @@
 
 sources = files('rte_cfgfile.c')
 headers = files('rte_cfgfile.h')
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/distributor/meson.build b/lib/distributor/meson.build
index cefe1b9f19..e1617fe389 100644
--- a/lib/distributor/meson.build
+++ b/lib/distributor/meson.build
@@ -9,3 +9,5 @@ else
 endif
 headers = files('rte_distributor.h')
 deps += ['mbuf']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/efd/meson.build b/lib/efd/meson.build
index 07fb1c29a7..e658dfc0a4 100644
--- a/lib/efd/meson.build
+++ b/lib/efd/meson.build
@@ -4,3 +4,5 @@
 sources = files('rte_efd.c')
 headers = files('rte_efd.h')
 deps += ['ring', 'hash']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/eventdev/meson.build b/lib/eventdev/meson.build
index 32abeba794..8d926eb326 100644
--- a/lib/eventdev/meson.build
+++ b/lib/eventdev/meson.build
@@ -29,3 +29,5 @@ headers = files(
 )
 deps += ['ring', 'ethdev', 'hash', 'mempool', 'mbuf', 'timer', 'cryptodev']
 deps += ['telemetry']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/fib/meson.build b/lib/fib/meson.build
index 5618c215a7..af3ae7fe53 100644
--- a/lib/fib/meson.build
+++ b/lib/fib/meson.build
@@ -56,3 +56,5 @@ if dpdk_conf.has('RTE_ARCH_X86_64') and binutils_ok.returncode() == 0
         endif
     endif
 endif
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/flow_classify/meson.build b/lib/flow_classify/meson.build
index a7a5b3d577..ce1211f6ab 100644
--- a/lib/flow_classify/meson.build
+++ b/lib/flow_classify/meson.build
@@ -4,3 +4,5 @@
 sources = files('rte_flow_classify.c', 'rte_flow_classify_parse.c')
 headers = files('rte_flow_classify.h')
 deps += ['net', 'table']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/graph/meson.build b/lib/graph/meson.build
index 6befb094d2..a038612420 100644
--- a/lib/graph/meson.build
+++ b/lib/graph/meson.build
@@ -12,3 +12,5 @@ sources = files(
 headers = files('rte_graph.h', 'rte_graph_worker.h')
 
 deps += ['eal']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/gro/meson.build b/lib/gro/meson.build
index e4fa2958bd..07d3219f5f 100644
--- a/lib/gro/meson.build
+++ b/lib/gro/meson.build
@@ -10,3 +10,5 @@ sources = files(
 )
 headers = files('rte_gro.h')
 deps += ['ethdev']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/gso/meson.build b/lib/gso/meson.build
index 622411df8f..3652e64200 100644
--- a/lib/gso/meson.build
+++ b/lib/gso/meson.build
@@ -11,3 +11,5 @@ sources = files(
 )
 headers = files('rte_gso.h')
 deps += ['ethdev']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/ip_frag/meson.build b/lib/ip_frag/meson.build
index ea2de09f75..92d1a3857e 100644
--- a/lib/ip_frag/meson.build
+++ b/lib/ip_frag/meson.build
@@ -11,3 +11,5 @@ sources = files(
 )
 headers = files('rte_ip_frag.h')
 deps += ['ethdev', 'hash']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/ipsec/meson.build b/lib/ipsec/meson.build
index 1497f573bb..3b491fe93d 100644
--- a/lib/ipsec/meson.build
+++ b/lib/ipsec/meson.build
@@ -7,3 +7,5 @@ headers = files('rte_ipsec.h', 'rte_ipsec_sa.h', 'rte_ipsec_sad.h')
 indirect_headers += files('rte_ipsec_group.h')
 
 deps += ['mbuf', 'net', 'cryptodev', 'security', 'hash']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/jobstats/meson.build b/lib/jobstats/meson.build
index 391e4f802e..0e7c2af930 100644
--- a/lib/jobstats/meson.build
+++ b/lib/jobstats/meson.build
@@ -3,3 +3,5 @@
 
 sources = files('rte_jobstats.c')
 headers = files('rte_jobstats.h')
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/kni/meson.build b/lib/kni/meson.build
index e2e4e44fc6..90631bcce3 100644
--- a/lib/kni/meson.build
+++ b/lib/kni/meson.build
@@ -8,3 +8,5 @@ endif
 sources = files('rte_kni.c')
 headers = files('rte_kni.h', 'rte_kni_common.h')
 deps += ['ethdev', 'pci']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/latencystats/meson.build b/lib/latencystats/meson.build
index 286558dd79..6f8540102e 100644
--- a/lib/latencystats/meson.build
+++ b/lib/latencystats/meson.build
@@ -4,3 +4,5 @@
 sources = files('rte_latencystats.c')
 headers = files('rte_latencystats.h')
 deps += ['metrics', 'ethdev']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/lpm/meson.build b/lib/lpm/meson.build
index 5ca54815e4..d6b7f890c9 100644
--- a/lib/lpm/meson.build
+++ b/lib/lpm/meson.build
@@ -13,3 +13,5 @@ indirect_headers += files(
 )
 deps += ['hash']
 deps += ['rcu']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/member/meson.build b/lib/member/meson.build
index 0e6b34c406..058584b19c 100644
--- a/lib/member/meson.build
+++ b/lib/member/meson.build
@@ -4,3 +4,5 @@
 sources = files('rte_member.c', 'rte_member_ht.c', 'rte_member_vbf.c')
 headers = files('rte_member.h')
 deps += ['hash']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/metrics/meson.build b/lib/metrics/meson.build
index 3cb43d9ee7..5b34c3c825 100644
--- a/lib/metrics/meson.build
+++ b/lib/metrics/meson.build
@@ -11,3 +11,5 @@ if jansson.found()
 endif
 
 deps += ['ethdev', 'telemetry']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/node/meson.build b/lib/node/meson.build
index 230aa6fa0f..1b05823e27 100644
--- a/lib/node/meson.build
+++ b/lib/node/meson.build
@@ -16,3 +16,5 @@ headers = files('rte_node_ip4_api.h', 'rte_node_eth_api.h')
 # Strict-aliasing rules are violated by uint8_t[] to context size casts.
 cflags += '-fno-strict-aliasing'
 deps += ['graph', 'mbuf', 'lpm', 'ethdev', 'mempool', 'cryptodev']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/pdump/meson.build b/lib/pdump/meson.build
index 3a95eabde6..5dfd211552 100644
--- a/lib/pdump/meson.build
+++ b/lib/pdump/meson.build
@@ -4,3 +4,5 @@
 sources = files('rte_pdump.c')
 headers = files('rte_pdump.h')
 deps += ['ethdev']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/pipeline/meson.build b/lib/pipeline/meson.build
index 9132bb517a..eb43707cc0 100644
--- a/lib/pipeline/meson.build
+++ b/lib/pipeline/meson.build
@@ -18,3 +18,5 @@ headers = files(
         'rte_swx_ctl.h',
 )
 deps += ['port', 'table', 'meter', 'sched', 'cryptodev']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/port/meson.build b/lib/port/meson.build
index 289bc47f6e..8d1c64d778 100644
--- a/lib/port/meson.build
+++ b/lib/port/meson.build
@@ -44,3 +44,5 @@ if dpdk_conf.has('RTE_LIB_KNI')
     headers += files('rte_port_kni.h')
     deps += 'kni'
 endif
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/rawdev/meson.build b/lib/rawdev/meson.build
index 30d363cd2d..f82c2f2529 100644
--- a/lib/rawdev/meson.build
+++ b/lib/rawdev/meson.build
@@ -5,3 +5,5 @@ sources = files('rte_rawdev.c')
 headers = files('rte_rawdev.h', 'rte_rawdev_pmd.h')
 
 deps += ['telemetry']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/regexdev/meson.build b/lib/regexdev/meson.build
index 7e12d8cd6d..b3ff541d1e 100644
--- a/lib/regexdev/meson.build
+++ b/lib/regexdev/meson.build
@@ -5,3 +5,5 @@ sources = files('rte_regexdev.c')
 headers = files('rte_regexdev.h', 'rte_regexdev_driver.h')
 indirect_headers += files('rte_regexdev_core.h')
 deps += ['mbuf']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/rib/meson.build b/lib/rib/meson.build
index 7bacbb4535..0fcdba0da9 100644
--- a/lib/rib/meson.build
+++ b/lib/rib/meson.build
@@ -5,3 +5,5 @@
 sources = files('rte_rib.c', 'rte_rib6.c')
 headers = files('rte_rib.h', 'rte_rib6.h')
 deps += ['mempool']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/sched/meson.build b/lib/sched/meson.build
index b24f7b8775..b066f6f0e9 100644
--- a/lib/sched/meson.build
+++ b/lib/sched/meson.build
@@ -9,3 +9,5 @@ headers = files(
         'rte_sched_common.h',
 )
 deps += ['mbuf', 'meter']
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/stack/meson.build b/lib/stack/meson.build
index 2f53f49677..47e9cc0297 100644
--- a/lib/stack/meson.build
+++ b/lib/stack/meson.build
@@ -5,8 +5,9 @@ sources = files('rte_stack.c', 'rte_stack_std.c', 'rte_stack_lf.c')
 headers = files('rte_stack.h')
 # subheaders, not for direct inclusion by apps
 indirect_headers += files(
-        'rte_stack_std.h',
-        'rte_stack_lf.h',
-        'rte_stack_lf_generic.h',
-        'rte_stack_lf_c11.h',
-)
+	'rte_stack_std.h',
+	'rte_stack_lf.h',
+	'rte_stack_lf_generic.h',
+	'rte_stack_lf_c11.h')
+build = false
+reason = 'not needed by SPDK'
diff --git a/lib/table/meson.build b/lib/table/meson.build
index b7b70b805e..c6af5eed1d 100644
--- a/lib/table/meson.build
+++ b/lib/table/meson.build
@@ -38,3 +38,5 @@ indirect_headers += files(
         'rte_lru_x86.h',
         'rte_table_hash_func_arm64.h',
 )
+build = false
+reason = 'not needed by SPDK'
diff --git a/meson.build b/meson.build
index 12cb6e0e83..ac3e3a10d2 100644
--- a/meson.build
+++ b/meson.build
@@ -65,7 +65,7 @@ subdir('drivers')
 
 # build binaries and installable tools
 subdir('usertools')
-subdir('app')
+#subdir('app')
 
 # build docs
 subdir('doc')
-- 
2.26.2

